shuffle/cli/src/transactions.rs

```diff
@@ -0,0 +1,227 @@
+        println!();
+        println!();
+    }
+
+    let mut old_seq_num = 0;
+    if !all_transactions.is_empty() {
+        old_seq_num = all_transactions
+            .get(all_transactions.len() - 1)
+            .ok_or_else(|| anyhow!("Couldn't get last transaction"))?["sequence_number"]
+            .to_string()
+            .replace('"', "")
+            .parse::<u64>()?;
+    }
+    if tailing {
+        // listening for incoming transactions
+        loop {
+            thread::sleep(time::Duration::from_millis(1000));
+            let resp = get_account_txns_response(&client, &path, old_seq_num, 100).await?;
+            let json_with_txns: serde_json::Value =
+                serde_json::from_str(resp.text().await?.as_str())?;
+            let size_of_json = json_with_txns
+                .as_array()
+                .ok_or_else(|| anyhow!("Couldn't convert to array"))?
+                .len();
+
+            // checking if there are transactions
+            if size_of_json > 0 {
+                let last_txn_seq_num = &json_with_txns
+                    .get(size_of_json - 1)
+                    .ok_or_else(|| anyhow!("Couldn't get last transaction"))?["sequence_number"]
+                    .to_string()
+                    .replace('"', "")
+                    .parse::<u64>()?;
+                old_seq_num = write_out_txns(
+                    json_with_txns,
+                    *last_txn_seq_num,
+                    old_seq_num,
+                    &mut io::stdout(),
+                )?;
+            }
+        }
+    }
+    Ok(())
+}
+
+pub fn write_out_txns<W: Write>(
+    json: Value,
+    last_txn_seq_num: u64,
+    old_seq_num: u64,
+    mut stdout: W,
+) -> Result<u64> {
+    // Only want to write the first transaction (seq num: 0) or
+    // transactions where the sequence number changes
+    if last_txn_seq_num == 0 || last_txn_seq_num > old_seq_num {
+        let all_transactions = json
+            .as_array()
+            .ok_or_else(|| anyhow::anyhow!("Failed to get transactions"))?;
+
+        // setting to 1 to exclude the index of the txn we've already printed out
+        let mut start_ind = 1;
+        if last_txn_seq_num == 0 {
+            // setting to 0 to include the first txn.
+            // Only needed for the very first txn
+            start_ind = 0;
+        }
+        for txn_index in start_ind..all_transactions.len() {
+            write_into(&mut stdout, &json[txn_index])?;
+            println!();
+            println!();
+        }
+    }
+    Ok(last_txn_seq_num)
+}
+
+pub async fn get_account_txns_response(
+    client: &Client,
+    path: &Url,
+    start: u64,
+    limit: u64,
+) -> Result<Response> {
+    Ok(client
+        .get(path.as_str())
+        .query(&[("start", start.to_string().as_str())])
+        .query(&[("limit", limit.to_string().as_str())])
+        .send()
+        .await?)
+}
+
+pub async fn get_sequence_number(
```

Comment:
nit: would rename to `get_account_sequence_number` to better disambiguate from txn sequence number.